pkg/cli/cmd/resource/list/list.go

```diff
@@ -148,25 +180,57 @@ func (r *Runner) Run(ctx context.Context) error {
 
 	var resourceList []generated.GenericResource
 
-	_, err = common.GetResourceTypeDetails(ctx, r.ResourceProviderNameSpace, r.ResourceTypeSuffix, client)
-	if err != nil {
-		return err
+	// If neither application name nor environment name is provided,
+	// and no resource type is specified, we can't list all resources
+	if r.ApplicationName == "" && r.EnvironmentName == "" && r.ResourceType == "" {
+		return clierrors.Message("Please specify a resource type, application name, or environment name")
 	}
 
-	if r.ApplicationName == "" {
-		resourceList, err = client.ListResourcesOfType(ctx, r.ResourceType)
+	// If no resource type is specified, but an environment name is provided,
+	// list all resources in the environment
+	if r.ResourceType == "" && r.EnvironmentName != "" {
```

Comment:
What if both environment and application names are provided? Should we filter resources for both? cc/ @willtsai

Reply:
yes, if both are provided, then filter for both.

Reply:
@ytimocin @willtsai - I've committed a change to filter when both environment and application names are provided, as well as adding new test cases. Let me know if I'm on the right track here :)